packages/vendure-plugin-accept-blue/src/api/accept-blue-service.ts

```diff
@@ -68,31 +79,186 @@ export class AcceptBlueService {
     const customer = await client.getOrCreateCustomer(
       order.customer.emailAddress
     );
+    if (!order.customer?.customFields?.activeBlueCustomerId) {
+      await this.customerService.update(ctx, {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
     // Save payment method
-    const paymentMethod = await client.getOrCreatePaymentMethod(
+    const paymentMethod = await client.getOrCreateCardPaymentMethod(
       customer.id,
       ccDetails
     );
-    return await this.payWithSavedPaymentMethod(
+    return await this.payWithCardPaymentMethodDetails(
+      ctx,
+      order,
+      amount,
+      client,
+      paymentMethod.id,
+      ccDetails
+    );
+  }
+
+  async payWithToken(
+    ctx: RequestContext,
+    order: Order,
+    amount: number,
+    client: AcceptBlueClient,
+    ccDetails: TokenPaymentMethodInput
+  ): Promise<HandlePaymentResult> {
+    if (!order.customer) {
+      throw new UserInputError(`Order must have a customer`);
+    }
+    const customer = await client.getOrCreateCustomer(
+      order.customer.emailAddress
+    );
+    if (!order.customer?.customFields?.activeBlueCustomerId) {
+      await this.customerService.update(ctx, {
+        id: order.customer?.id,
+        customFields: { activeBlueCustomerId: customer.id },
+      });
+    }
+    // Save payment method
+    const methods = await client.getPaymentMethods(
```

Comment:
Comment doesn't seem correct, this only checks if a payment method with the token already exists right?

Also: Is a payment method already created when we have a tokenized card? I don't fully understand how it works yet.